File: 05_arrays/05_03_check_if_array_is_sorted_and_rotated/check_if_array_is_sorted_and_rotated.cpp
#include <bits/stdc++.h>

using namespace std;

bool isSortedAndRotated(int arr[], int n) {
    int count = 0;

    for (size_t i = 1; i < n; i++)
    {
        if (arr[i-1] > arr[i])
        {
            count++;
        }
    }
    
    if (arr[0] < arr[n - 1])
    {
        count++;
    }
    
    return count <= 1;
}

int main() {
    int arr[] = { 3, 4, 5, 1, 2 };
    int n = sizeof(arr) / sizeof(arr[0]);

    cout << "Elements:-" << endl;
    for (size_t i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;

    cout << "Is Sorted: " << isSortedAndRotated(arr, n);
}

File: 05_arrays/05_03_check_if_array_is_sorted_and_rotated/check_if_array_is_sorted_and_rotated.py
def check(nums: list) -> bool:
    count = 0
    for i in range(1, len(nums)):
        if (nums[i] < nums[i-1]):
            count += 1

    if nums[0] < nums[len(nums) - 1]:
        count += 1
    
    return count <= 1

arr = [3, 4, 5, 1, 2]
print("Elements:-")
print(arr)
print(f"Is sorted: {check(arr)}")